distribution/graal/oal-rt-graal-native/src/main/java/org/apache/skywalking/oal/rt/OALRuntime.java

```diff
@@ -0,0 +1,673 @@
+        } else {
+            startByJar(currentClassLoader);
+        }
+        INITIALED = true;
+    }
+
+    private void startByNativeImage(ClassLoader currentClassLoader) {
+        try (FileSystem  fileSystem = FileSystems.newFileSystem(URI.create("resource:/"), Map.of(), currentClassLoader)) {
+            Path metrics = fileSystem.getPath("org/apache/skywalking/oap/server/core/source/oal/rt/metrics");
+            Path dispatcher = fileSystem.getPath("org/apache/skywalking/oap/server/core/source/oal/rt/dispatcher");
+
+            try (java.util.stream.Stream<Path> files = Files.walk(metrics)) {
+                files.forEach(file -> {
+                    if (!file.toString().endsWith(".class")) {
+                        return;
+                    }
+                    String name = file.toString().replace(File.separator, ".");
+
+                    name = name.substring(0, name.length() - ".class".length());
+                    if (name.startsWith("org.apache.skywalking.oap.server.core.source.oal.rt.metrics.builder")) {
+                        return;
+                    }
+                    try {
+                        Class<?> aClass = Class.forName(name);
+                        if (!aClass.isAnnotationPresent(Stream.class)) {
+                            return;
+                        }
+                        metricsClasses.add(aClass);
+                    } catch (ClassNotFoundException e) {
+                        // should not reach here
+                        log.error(e.getMessage());
+                    }
+                });
+            }
+            catch (IOException e) {
+                log.error("Failed to walk the path: " + metrics, e);
+            }
+            try (java.util.stream.Stream<Path> files = Files.walk(dispatcher)) {
+                files.forEach(file -> {
+                    if (!file.toString().endsWith(".class")) {
+                        return;
+                    }
+                    String name = file.toString().replace(File.separator, ".");
+                    name = name.substring(0, name.length() - ".class".length());
+                    try {
+                        dispatcherClasses.add(Class.forName(name));
+                    } catch (ClassNotFoundException e) {
+                        // should not reach here
+                        log.error(e.getMessage());
+                    }
+                });
+            }
+            catch (IOException e) {
+                log.error("Failed to walk the path: " + dispatcher, e);
+            }
+        } catch (IOException e) {
+            // should not reach here
+            log.error("Failed to create FileSystem", e);
+        }
+    }
+
+    private void startByJar(ClassLoader currentClassLoader) {
+        String metricsPath = "org/apache/skywalking/oap/server/core/source/oal/rt/metrics";
+        String dispatcherPath = "org/apache/skywalking/oap/server/core/source/oal/rt/dispatcher";
```

Comment:
These could be converted from `METRICS_FUNCTION_PACKAGE`... by codes. Don't hard code things in two places.